### DISCUSSION TITLE
Cannot start the LS after updating nvim and plugins

### DISCUSSION THREAD

**Discussion by ChrisAmelia** [mfussenegger/nvim-jdtls#497] [Q&A] [answered]
Command to start the server
```
java -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Dlog.protocol=true -Dlog.level=ALL --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED -jar /home/user/GitHub/eclipse.jdt.ls/org.eclipse.jdt.ls.product/target/repository/plugins/org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar -configuration /home/user/GitHub/eclipse.jdt.ls/org.eclipse.jdt.ls.product/target/repository/config_linux/ -data /home/user/my-workspace
```
  
When running in nvim, this is the result:

> Client 1 quit with exit code 13 and signal 0

`java --version`

```txt
openjdk 17.0.7 2023-04-18
OpenJDK Runtime Environment (build 17.0.7+7-Ubuntu-0ubuntu118.04)
OpenJDK 64-Bit Server VM (build 17.0.7+7-Ubuntu-0ubuntu118.04, mixed mode, sharing)
```

`:LspLog`

```txt
[START][2023-05-28 17:54:21] LSP logging initiated
[ERROR][2023-05-28 17:54:21] .../vim/lsp/rpc.lua:734	"rpc"	"java"	"stderr"	"WARNING: Using incubator modules: jdk.incubator.vector, jdk.incubator.foreign\n"
[WARN][2023-05-28 17:54:21] .../lua/vim/lsp.lua:97	"method textDocument/documentSymbol is not supported by any of the servers registered for the current buffer"
```

Config:

```lua
local capabilities = require('cmp_nvim_lsp').default_capabilities()
capabilities.textDocument.completion.completionItem.snippetSupport = true
capabilities.textDocument.completion.completionItem.preselectSupport = true
capabilities.textDocument.completion.completionItem.insertReplaceSupport = true
capabilities.textDocument.completion.completionItem.labelDetailsSupport = false
capabilities.textDocument.completion.completionItem.deprecatedSupport = true
capabilities.textDocument.completion.completionItem.commitCharactersSupport = true
capabilities.textDocument.completion.completionItem.tagSupport = { valueSet = { 1 } }
capabilities.semanticTokensProvider = nil
capabilities.textDocument.completion.completionItem.resolveSupport = {
	properties = {
		'documentation',
		'detail',
		'additionalTextEdits',
	}
}

cmd = {
	'java', -- or '/path/to/java11_or_newer/bin/java'
	
	'-Declipse.application=org.eclipse.jdt.ls.core.id1',
	'-Dosgi.bundles.defaultStartLevel=4',
	'-Declipse.product=org.eclipse.jdt.ls.core.product',
	'-Dlog.protocol=true',
	'-Dlog.level=ALL',
	'-Xms1g',
	'--add-modules=ALL-SYSTEM',
	'--add-opens', 'java.base/java.util=ALL-UNNAMED',
	'--add-opens', 'java.base/java.lang=ALL-UNNAMED',
	'-jar', '/home/user/GitHub/eclipse.jdt.ls/org.eclipse.jdt.ls.product/target/repository/plugins/org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar',
	'-configuration', '/home/user/GitHub/eclipse.jdt.ls/org.eclipse.jdt.ls.product/target/repository/config_linux/',

	'-data', '/home/user/my-workspace'
},

root_dir = require('jdtls.setup').find_root({'.git', 'mvnw', 'gradlew'}),

settings = {
	java = {
		autobuild = { enabled = false };
                                                   
		inlayHints = {
			parameterNames = {
				enabled = "all",
			},
		},
                                                   
		referencesCodeLens = { enabled = true };
                                                   
		signatureHelp = { enabled = true };
                                                   
		configuration = {
			updateBuildConfiguration = "disabled",
			maven = { userSetting = nil }
		};
                                                   
		import = {
			maven = { enabled = false },
			exclusions = {
				"**/node_modules/**",
				"**/.metadata/**",
				"**/archetype-resources/**",
				"**/META-INF/maven/**",
				"/**/test/**"
			},
		};
                                                   
		completion = {
			importOrder = {
				"java",
				"javax",
				"com",
				"org"
			},
		};
	}
},

capabilities = capabilities,

init_options = {
	bundles = {}
},
```

**Comment by ChrisAmelia** [comment, marked as answer]
tl;dr use `~` instead of `/home/user/`.

https://github.com/eclipse/eclipse.jdt.ls/issues/2672
